lib/compiler/src/beam_validator.erl

```diff
@@ -1403,21 +1405,37 @@ validate_bs_start_match({f,Fail}, Live, Slots, Src, Dst, Vst) ->
 %%
 %% Common code for validating bs_get* instructions.
 %%
-validate_bs_get(Op, Fail, Ctx, Live, Stride, Type, Dst, Vst) ->
-    assert_no_exception(Fail),
-
+validate_bs_get(Op, Fail, Ctx, Live, Stride, ValueType, Dst, Vst) ->
     assert_type(#t_bs_context{}, Ctx, Vst),
-    verify_live(Live, Vst),
-    verify_y_init(Vst),
 
+    %% The tail unit will not be tested, so we try to retain unit information
+    %% by taking the GCD of our context's unit and the increments we know the
+    %% match will advance by.
     #t_bs_context{tail_unit=TailUnit} = get_raw_type(Ctx, Vst),
     CtxType = #t_bs_context{tail_unit=gcd(Stride, TailUnit)},
 
+    validate_bs_get_1(Op, Fail, Ctx, Live, CtxType, ValueType, Dst, Vst).
+
+validate_bs_get_all(Op, Fail, Ctx, Live, Stride, ValueType, Dst, Vst) ->
+    assert_type(#t_bs_context{}, Ctx, Vst),
+
+    %% Set the tail unit to the stride of the current operation to match the
+    %% SSA pass. This is safe as the tail unit is irrelevant at the very end.
+    CtxType = #t_bs_context{tail_unit=Stride},
```

Comment:
This is more of a band-aid than an actual fix. We need to track the units for saved positions, but that's a bit too complicated for an emergency patch. I'll try to get this fixed in OTP 23.2.